app/gui/view/graph-editor/src/lib.rs

```diff
@@ -1681,6 +1685,79 @@ impl GraphEditorModel {
+        let mut min_distance_squared = f32::MAX;
+        let mut nearest_node = None;
+        let nodes = self.nodes.all.raw.borrow();
+        for node in nodes.values() {
+            // For calculating the approximate distance between a node and the mouse pointer, one
+            // of several possible reference points on a node can be chosen. In order of increasing
+            // complexity and precision:
+            //  a) node.position(),
+            //  b) the geometrical center of the node's area,
+            //  c) a point closest to the mouse pointer on the line segment connecting node's
+            //     leftmost & rightmost points,
+            //  d) a point closest to the mouse pointer on the bounding box of the node
+            //     (with extra handling for when the mouse is inside the bounding box),
+            //  e) a point closest to the mouse pointer on the rounded rectangle forming the node's
+            //     shape (with extra handling for when the mouse is inside the rounded rectangle).
+            //  The following code implements variant (c), the first of the listed ones that
+            //  resulted in a UX which feels reasonably predictable and intuitive.
+            let node_position = node.position().xy();
+            let node_rightmost_x = node_position.x + node.model.width();
+            let mouse_x = mouse_position.x;
+            let node_point_near_mouse = if mouse_x <= node_position.x {
+                node_position
+            } else if mouse_x < node_rightmost_x {
+                Vector2(mouse_x, node_position.y)
+            } else {
+                Vector2(node_rightmost_x, node_position.y)
+            };
+            let distance_squared = (node_point_near_mouse - mouse_position).norm_squared();
+            if distance_squared < min_distance_squared {
+                min_distance_squared = distance_squared;
+                nearest_node = Some(node.clone_ref());
+            }
+        }
+        nearest_node
+    }
+
+    pub fn find_place_dictated_by_node(
+        &self,
+        node: Node,
+        mouse_position: Vector2,
+    ) -> Option<Vector2> {
+        let node_position = node.position();
+        let node_left = node_position.x;
+        let node_right = node_position.x + node.model.width();
+        let node_top = node_position.y + node.model.height() / 2.0;
+        let node_bottom = node_position.y - node.model.height() / 2.0;
+        let area_left = node_left - NODE_PLACEMENT_AREA_SIDE;
+        let area_right = node_right + NODE_PLACEMENT_AREA_SIDE;
+        let area_top = node_top + NODE_PLACEMENT_AREA_ABOVE;
+        let area_bottom = node_bottom - NODE_PLACEMENT_AREA_BELOW;
+        let placement_area = selection::BoundingBox::from_corners(
+            Vector2(area_left, area_top),
+            Vector2(area_right, area_bottom),
+        );
+        placement_area.contains(mouse_position).then(|| self.find_free_place_under(node.id()))
+    }
+
```

Comment:
The name of functions in the block above does not intuitively say what the function does (except `find_nearest_node`). For example:`find_place_by_dropping_edge` suggests, that we drop edge to find position. We also don't quite "find" the position, given that in most cases the position is granted to us (as the `mouse_position`).

`find_place_dictated_by_node` might be ` should be rather `node_position_based_on_mouse_cursor` (and return the mouse position instead of `None`)

Reply:
Changed the names to the ones we discussed on a call - thanks for the help with inventing better names! ❤️